pkg/ottl/ottlfuncs/README.md

```diff
@@ -91,13 +92,97 @@ Examples:
 
 - `delete_matching_keys(resource.attributes, "(?i).*password.*")`
 
+### flatten
+
+`flatten(target, prefix, depth)`
```

Comment:
How do you feel about `flatten(target, Optional[prefix, depth])` or `flatten(target, Optional[prefix], Optional[depth])`

Reply:
Between those two I'd prefer the second one. I think ultimately it would be nice to just replicate the Go function signatures here with types and everything, but that can come later.

Reply:
Updated